lib/spack/spack/directives.py

```diff
@@ -180,24 +180,43 @@ def _depends_on(pkg, spec, when=None):
         when = pkg.name
     when_spec = parse_anonymous_spec(when, pkg.name)
 
+    if deptypes is None:
+        # The default deptype is build and link because the common case is to
+        # build against a library which then turns into a runtime dependency
+        # due to the linker.
```

Comment:
If I am not mistaken `run` is thereby used exclusively in `spack load` and should not affect environment of `spack install` at all? I think example you mentioned will be helpful for package developers to understand the difference. So i would mention it in the description.